### PR TITLE
Clarify Tag Length Setting in OCB Mode

### LINKED ISSUE #8331: EVP_CTRL_AEAD_SET_TAG fails for OCB
When **decrypting** using OCB mode, setting a custom tag size fails unless `EVP_CTRL_AEAD_SET_TAG` is called twice (first with `NULL`, and then the second time with the actual tag buffer).

This fails unexpectedly when **decrypting**:
```c
assert(EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG, tag_size, tag));
```

This succeeds when **decrypting**:
```c
assert(EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG, tag_size, NULL));
assert(EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG, tag_size, tag));
```

The documentation indicates that this double call should only be necessary when **encrypting**, not when **decrypting**:

> In OCB mode, calling this before encryption with tag set to NULL sets
the tag length. If this is not called prior to encryption, a default tag
length is used.

I think this is probably a bug?

### PR DESCRIPTION
**Pull request by erbsland-dev**
Fixes #8331: Updated the description for setting the tag length in OCB mode to remove the misleading “when encrypting” and "during encryption" phrasing. This change emphasizes that setting a custom tag length requires a call with NULL, applicable to both encryption and decryption contexts.

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/EVP_EncryptInit.pod
Comment (t8m) [inline doc/man3/EVP_EncryptInit.pod:1440]: Actually also remove the `during encryption` here

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by fwh-dc** [comment]
@t8m Will the mentioned issue be auto-closed when `Fixes #8331` is not on its own line in the commit message? If not It'd be great if you (@erbsland-dev) could update the commit message.

**Comment by t8m** [comment]
Not sure. At worst we can either reword the commit message when merging or close the fixed issue manually.

**Comment by paulidale** [comment]
The two are linked already.  Merging & closing this will close #8331 too.

**Review by fwh-dc — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
> The two are linked already. Merging & closing this will close #8331 too.

Actually just linking the issue and PR in the Github interface does not help with closing the issue as the merge happens outside of GH and the commit is rebased before merging so GH cannot associate it with the PR from which it comes. So yeah, sometimes the committer who does the merge needs to remember to close the linked issue manually.

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for your contribution.

**Pull request closed without merging**
